data/pigui/modules/radar.lua

```diff
@@ -304,27 +390,78 @@ local function displayRadar()
 		toggle_radar = true
 	end
 
-	-- Handle mouse if it is in the radar area
-	local mp = ui.getMousePos()
-	-- TODO: adjust properly for 3D radar; bit more challenging as its an ellipse
-	if (mp - center):length() < radar2d.getRadius(radar2d) then
-		ui.popup("radarselector", function()
-			if ui.selectable(lui.HUD_2D_RADAR, shouldDisplay2DRadar, {}) then
+	--
+	-- popup functions
+	--
+	ui.popup("radarselector", function()
+		if ui.selectable(lui.HUD_2D_RADAR, shouldDisplay2DRadar, {}) then
+			if not shouldDisplay2DRadar then
 				toggle_radar = true
 			end
-			if ui.selectable(lui.HUD_3D_RADAR, not shouldDisplay2DRadar, {}) then
+			radar_popup_displayed = false
+		end
+		if ui.selectable(lui.HUD_3D_RADAR, not shouldDisplay2DRadar, {}) then
+			if shouldDisplay2DRadar then
 				toggle_radar = true
 			end
-		end)
+			radar_popup_displayed = false
+		end
+	end)
+	ui.popup("radartargetselector", function()
+		for k,v in pairs(popup_targets) do
+			if ui.selectable(v.label) then
+				onTargetClicked(v)
+				radar_popup_displayed = false
+			end
+		end
+	end)
 
-		if ui.isMouseClicked(1) then
+	--
+	-- mouse handling
+	--
+	local isMouseOverRadar = function()
+		if shouldDisplay2DRadar then
+			local mp = ui.getMousePos()
+			return (mp - center):length() < radar2d.getRadius(radar2d)
+		end
+		return ui.isMouseHoveringRect(
+			Vector2(center.x - ui.reticuleCircleRadius * 0.9,
+			        center.y - ui.reticuleCircleRadius * 0.7),
+			Vector2(center.x + ui.reticuleCircleRadius * 0.9,
+			        center.y + ui.reticuleCircleRadius * 0.7))
```

Comment:
Should probably move the window size definitions (below) to the top, and re-use the sizes here.

Reply:
Fixed in 88cc619fbcde8c5b82620aece290ce7b8e3139d6